src/mame/adp/stella8085.cpp

```diff
@@ -124,40 +129,53 @@ void stella8085_state::machine_start()
 	save_item(NAME(m_digit));
 }
 
-void stella8085_state::program_map(address_map &map)
+void stella8085_state::program_4040_map(address_map &map)
 {
-	map(0x0000, 0x8fff).rom(); // ICE6, ICD6, ICC5
-	map(0x9000, 0x933f).rw("rtc", FUNC(mc146818_device::read_direct), FUNC(mc146818_device::write_direct));
+	map(0x0000, 0x4fff).rom();
+	map(0x5000, 0x5fff).ram();
+	map(0x6000, 0x6fff).ram();
+	//map(0x6000, 0x603f).rw("rtc", FUNC(mc146818_device::read_direct), FUNC(mc146818_device::write_direct));
+	map(0x7000, 0x7fff).rom();
+	map(0x8000, 0x9fff).ram();
+	map(0xa000, 0xffff).ram();
+}
+
+void stella8085_state::program_4087_map(address_map &map)
+{
+	map(0x0000, 0x3fff).rom(); // ICE6
+	map(0x4000, 0x7fff).rom(); // ICD6
+	map(0x8000, 0x8fff).rom(); // ICC5
+	map(0x9000, 0x903f).rw("rtc", FUNC(mc146818_device::read_direct), FUNC(mc146818_device::write_direct));
 	map(0xa000, 0xafff).ram(); // ??
 	map(0xc000, 0xc7ff).ram(); // ICC6
 }
 
-void stella8085_state::large_program_map(address_map &map)
+void stella8085_state::program_4109_map(address_map &map)
 {
 	map(0x0000, 0x7fff).rom(); // ICE6
 	map(0x8000, 0x9fff).ram(); // ICC6
 	map(0xa000, 0xffff).rom(); // ICD6
 }
 
-void stella8085_state::small_program_map(address_map &map)
+void stella8085_state::io_4040_map(address_map &map)
 {
-	map(0x0000, 0x4fff).rom();
-	map(0x5000, 0x5fff).ram();
-	map(0x6000, 0x633f).rw("rtc", FUNC(mc146818_device::read_direct), FUNC(mc146818_device::write_direct));
-	map(0x7000, 0x7fff).rom();
+	map(0x00, 0x00).w(FUNC(stella8085_state::io00));
+	map(0x70, 0x73).rw("ppi", FUNC(i8255_device::read), FUNC(i8255_device::write));
+	map(0x80, 0x81).rw("kdc", FUNC(i8279_device::read), FUNC(i8279_device::write));
+	map(0x90, 0x9f).rw("muart", FUNC(i8256_device::read), FUNC(i8256_device::write));
```

Comment:
Remember it’s better to use object finders rather than literal tags if you have them anyway.